pkg/operator/observe_config.go

```diff
@@ -62,18 +62,47 @@ func NewConfigObserver(
 // sync reacts to a change in prereqs by finding information that is required to match another value in the cluster. This
 // must be information that is logically "owned" by another component.
 func (c ConfigObserver) sync() error {
+
+	observedConfig := map[string]interface{}{}
+	var err error
+
+	if observedConfig, err = c.observeEtcdEndpoints(observedConfig); err != nil {
```

Comment:
we don't do this.  If you are assigning to a value that escapes the block, you don't do it in the block.